File: solutions/cpp/Count Good Nodes in Binary Tree.cpp
#include <stdio.h>
using namespace std;
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    int goodNodes(TreeNode *root)
    {
        return Dfs(root, root->val);
    }

    int Dfs(TreeNode *root, int val)
    {
        if (root == NULL)
        {
            return 0;
        }
        int res = 0;
        if (root->val >= val)
        {
            val = root->val;
            res = 1;
        }
        return res + Dfs(root->left, val) + Dfs(root->right, val);
    }
};

File: solutions/python/Count Good Nodes in Binary Tree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        return self.Dfs(root,root.val);
    def Dfs(self,root,val):
        if not root:
            return 0;
        res=1 if root.val >= val else 0;
        val=root.val if(root.val>=val) else val;
        return res+self.Dfs(root.left,val)+self.Dfs(root.right,val);